ApiTests/Tests/Pages/DynamicControlPage.cs

```diff
@@ -0,0 +1,47 @@
+﻿using OpenQA.Selenium;
+using OpenQA.Selenium.Internal;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TestProject.Utils;
+
+namespace TestProject.Tests.Pages
+{
+    internal class DynamicControlsPage
+    {
+        private static readonly By dynamicControl0 = By.XPath(string.Format(XpathPatterns.preciseTextXpath, "Dynamic Controls"));
+        private static readonly By dynamicControl = By.XPath(string.Format(XpathPatterns.dynamicControl, "Dynamic Controls"));
+        private static readonly By enableBtn = By.XPath(string.Format(XpathPatterns.enableBtn1, "enable button"));
+        private static readonly By inputField = By.XPath(string.Format(XpathPatterns.inputField1, "input"));
+        private static readonly string randomValue = Guid.NewGuid().ToString();
+        string text = "automation class";
+        public void ClickOnEnable()
+        {
+            Browser.GetDriver().FindElement(enableBtn).Click();
+            Thread.Sleep(5000);
+        }
+        //input randomly generated text
+        public void enterText()
+        {
+            Browser.GetDriver().FindElement(inputField).Clear();
+            Browser.GetDriver().FindElement(inputField).SendKeys(text+randomValue);
+        }
+        //assert input is enabled
+        public void assertInputIsEnabled()
+        {
+            var enable_diabledBtnMsg = Browser.GetDriver().FindElement(enableBtn).Text;
+            Assert.That(enable_diabledBtnMsg, Is.EqualTo("Disable"));
```

Comment:
assertions should not be placed in a page class, they should be present only in test method (as a part of a test class)
In this case I'd suggest to change method to public bool IsInputEnabled, so that it would return true or false based on the status of the element and we'd use that info in test :)